### PR TITLE
Improve parser and change expiry date to LocalDate

### LINKED ISSUE #23: Improve Parser
Add more OOP to Parser
Accept only valid dates for expiry date

### PR DESCRIPTION
**Pull request by bernardboey**
Add more OOP to Parser
Accept only valid dates for expiry date
Fixes #23

### CODE REVIEW

File: src/main/java/expiryeliminator/commands/AddIngredientCommand.java [hunk 1]
```diff
@@ -19,7 +21,7 @@
 
     private final Ingredient ingredient;
 
-    public AddIngredientCommand(String name, int quantity, String expiryDate) {
+    public AddIngredientCommand(String name, int quantity, LocalDate expiryDate) {
```
Comment (KairuiHu) [inline src/main/java/expiryeliminator/commands/AddIngredientCommand.java:24]: It is a good practice to declare the type of date as LocalDate

**Review by KairuiHu — APPROVED** [review]
Good to make the code more flexible

**Comment by vincentlauhl** [comment]
Really appreciate your effort in making the code neater. I think for the prefix it isn't really necessary to have a class of its own since you're only getting the String, unless you have other plans for it later on?

**Comment by KairuiHu** [comment]
You did a good job in improving the structure of the code to make the logic clearer and more organized. Meanwhile, you managed to make the code more OOP, which also helps the rest of the teammates to have a better understanding of the meaning and flow of the whole program. Perhaps, more parser classes could be added in future work like unit parser which enables users to update units for ingredients.

**Pull request merged**